### PR TITLE
Converts v1 devfiles to v2

### LINKED ISSUE #3142: Migrate the devfile v1 Tests to devfile  v2
/kind test

Followup issue for PR https://github.com/openshift/odo/pull/3216

### PR DESCRIPTION
**Pull request by mik-dass**
**What type of PR is this?**

/kind code-refactoring

**What does does this PR do / why we need it**:

This PR converts the v1 devfiles to v2

**Which issue(s) this PR fixes**:

Fixes #3142 

**How to test changes / Special notes to the reviewer**:

- Tests should pass

[NOTE] This PR is raised against parser-v2 branch for testing. Once parser-v2 is merged into master, the target branch will be changed to master.

### CODE REVIEW

**Comment by mik-dass** [comment]
```
• Failure [317.853 seconds]
odo devfile watch command tests
/go/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_watch_test.go:15
  when executing odo watch after odo push with custom commands
  /go/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_watch_test.go:115
    should listen for file changes [It]
    /go/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_watch_test.go:116
    Timeout out after � minutes
    /go/src/github.com/openshift/odo/tests/helper/helper_generic.go:148
```
retest

**Comment by mik-dass** [comment]
Due to https://github.com/openshift/odo/issues/3308, devfiles with volume Mounted containers are failing.

File: tests/examples/source/devfiles/nodejs/devfile-with-volumes.yaml [hunk 1]
```diff
@@ -1,53 +1,52 @@
...
+      mountSources: true
+      volumes:
+        - name: myvol
+          containerPath: /data
+  - container:
+      name: runtime2
+      image: quay.io/eclipse/che-nodejs10-ubi:nightly
+      memoryLimit: 1024Mi
+      mountSources: false
+      volumes:
```
Comment (adisky) [inline tests/examples/source/devfiles/nodejs/devfile-with-volumes.yaml:31]: volume needs to specified like this
https://github.com/elsony/devfile2-registry/blob/master/devfiles/eclipse/springBoot/1.0.0/devfile.yaml#L20

Comment (mik-dass) [reply tests/examples/source/devfiles/nodejs/devfile-with-volumes.yaml]: fixed

**Comment by maysunfaisal** [comment]
@mik-dass any estimate when this is expected to go in? I was working on PR https://github.com/openshift/odo/pull/3291 and was wondering if i have to do anything for v2 devfile integration tests

**Comment by mik-dass** [comment]
> @mik-dass any estimate when this is expected to go in? I was working on PR #3291 and was wondering if i have to do anything for v2 devfile integration tests

Since the spring boot tests cannot be ported right now with the current schema, I think the PR is ready for review.

**Comment by adisky** [comment]
> > @mik-dass any estimate when this is expected to go in? I was working on PR #3291 and was wondering if i have to do anything for v2 devfile integration tests
> 
> Since the spring boot tests cannot be ported right now with the current schema, I think the PR is ready for review.

I think spring-boot can be ported too without updating the schema, if the reason is `command` `args` then i remember i tested v2 springboot and was able to pass `command` `args` too, without updating schema.

**Comment by mik-dass** [comment]
> I think spring-boot can be ported too without updating the schema, if the reason is command args then i remember i tested v2 springboot and was able to pass command args .

OK let me test it

**Comment by mik-dass** [comment]
> OK let me test it

Tests having a init command won't work with V2 as init command group type is not defined for V2.

**Comment by adisky** [comment]
> @mik-dass yes init command is currently broken with master branch when using v2 devfiles.
> 
> But i think most of the devfiles from https://github.com/odo-devfiles/registry should be working incl spring. I also see that we have devfiles2 dir in the examples/source dir.. Should we be updating those instead? Because we are currently converting some to v2 and leaving the init devfiles as v1 and then we have a devfiles2 dir.
> 
> Also maybe we should disable the init tests and open an issue and tag with main issue #2936? Because with v2 init is broken.
> 
> Or maybe I can do the changes with #3291 since I am completely removing devfile v1 registry and the tests are passing on this PR.

@mik-dass @maysunfaisal We are not intended to support init command with V2. same functionality would be achieved by using events.
AFAIK `init` would be removed with V1, so I guess it is ok to remove/comment init tests and as @maysunfaisal said tag it with #2936.

IMO If we want to keep init devfile for reference, we should be move them to /devfileV1 folder, so devfileV1 folder would have files that cannot be converted into V2.

**Comment by mik-dass** [comment]
@adisky @maysunfaisal OK I will move the init devfiles to a devfilesV1 folder and remove the devfilesv2 folder as those devfiles are already in the devfiles folder now.

**Comment by mik-dass** [comment]
```
  * could not initialize namespace: could not create role binding for: rolebindings.rbac.authorization.k8s.io is forbidden: User "system:serviceaccount:ci:ci-operator" cannot create resource "rolebindings" in API group "rbac.authorization.k8s.io" in the namespace "ci-op-h97ympkk"
time="2020-06-11T07:15:59Z" level=info msg="Reporting job state \"failed\" with reason \"initializing_namespace\""
```
/retest

**Comment by mik-dass** [comment]
> we should open an issue for the tests and remind the person working on init containers that they should be removed at the time v2 devfiles work. @mik-dass

You mean remind them to move the init tests to v1 folder?

**Comment by adisky** [comment]
@mik-dass We should log an additional issue for `init` tests, as the related issue would get closed with this PR 
`Migrate init functionality tests from devfile V1 to devfile V2 once, the functionality has been implemented by devfile V2 `

**Comment by mik-dass** [comment]
> Migrate init functionality tests from devfile V1 to devfile V2 once the functionality has been implemented by devfile V2

Are there even plans upstream to add it?

**Comment by adisky** [comment]
> > Migrate init functionality tests from devfile V1 to devfile V2 once the functionality has been implemented by devfile V2
> 
> Is there any plan to add it in the upstream schema?

 `init` directly would not be added but there would be something to achieve same functionality.

**Comment by mik-dass** [comment]
> init directly would not be added but there would be something to achieve same functionality.

@kadel Any plans for something equivalent to init in v2?

**Comment by maysunfaisal** [comment]
@mik-dass yes, post start and post stop events would achieve the same functionality as init

**Comment by mik-dass** [comment]
> @mik-dass yes, post start and post stop events would achieve the same functionality as init

Done https://github.com/openshift/odo/issues/3353

**Pull request merged**